### ISSUE TITLE
DestinyScript support

### ISSUE THREAD

**Issue by Jalax6** [EasyRPG/Player#1227]
Hello!

Was wondering about some coding stuff to be supported by Easy RPG - more specifically, the DestinyScript.

http://bananen-joe.de/DestinyDLL/?Language=1

It's an open-source patch for the RPG Maker 2k wich is really powerful, allowing us to change the content of already rendered image files, reading external files, string support, pointers support for every command, full keyboard and even full data access trought the engine.

It's triggered trough the Comment command. By adding a dollar sign on the start of the comment, every letter after will be interpreted and executed as script. Even being powerful as that, there are some limitations the original editor itself puts on it.

At start, I used the RPG Maker 2009 - made by Cherry, same as PicPointer -  that allows me to use giant comment boxes up to 100 lines, but sometimes it still not enough.

Actually, lcu2xml is already a GREAT addition to my developing tools due to the possibility of writing giant scripts by editing direcly the comments and converting back!

However, EasyRPG binary is a lot more stable than the old RPG_RT (mainly for the most recent OS), and it would be really really nice to take that add-on as native in EasyRPG.

**Comment by carstene1ns** [comment]
Is this the same as "Destiny Patch" by Cherry or are these unrelated? (→ #1183)

**Comment by Jalax6** [comment]
Cherry didn’t make these, also that are the 1.0 version, the scripting tools are poor compared to the 2.0 version on the website I linked.  <https://github.com/notifications/beacon/AThxilrlf1b2dmektkEt02jQMx06stQGks5sRkUHgaJpZM4OjBuR.gif>

**Comment by rohkea** [comment]
This was discussed in the IRC some time ago, see #1183 

DestinyScript looks like a subject of JavaScript, so we could use duktape or other JS interpreter and just write a compatible API for it. However, it might take quite some time to implement as few games use DestinyScript so far (or maybe we don’t know about them?).

**Comment by Jalax6** [comment]
Yeah, most of the games using this patch are Brazillian games (huehue BR), and don’t think they are strict-js since there are a lot of interesting commands there.

 

AuroraSheet, for example, by direct translation of the help file in german:

 

“With the AuroraSheet object within the Picture/Charset/Chipset [], the graphic can be accessed and eddited as well”
 

Basically do stuff like one-image hud with bars, or Just a font-file that literally writes text, all on a blank Picture that keeps recieving new data.

 

Plus, destiny script has pallete edition, server connection/creation, external file reading, fullscreen and lot more.

 

I thought that the source code would simplify the proccess, but looks like I was wrong lol

 

Anyway, going to use XML converter and try to enhance the original RPG_RT for better compatibility anyway.

**Comment by rohkea** [comment]
> and don’t think they are strict-js since there are a lot of interesting commands there

The language described here http://cherrytree.at/misc/destiny/DestinyScript_Eng/ looks like a subset of JS. Is there something beyond this description?

> most of the games using this patch are Brazillian games

Maybe you know where we can get a list of games using this patch? This would be useful for testing in the future (although please don’t get your hopes too high since this is a lot of work so even if we do this, it will take quite some time).

**Comment by Ghabry** [comment]
I think this is a duplicate of #1183 because I actually meant DestinyScript. Though this one has much more discussion then the other issue :D

> Just came up with it because bananen-joe has the full source code for that....

Afaik the source is written in assembler (he never finished the C version) which makes it quite useless for us because ASM is hard to understand.

And we discussed using Duktape (a JS-interpreter) for DestinyScript because the syntax looks like valid Javascript which means we don't have to do the parsing on our own (which saves lots of dev-time)

**Comment by Jalax6** [comment]
> Maybe you know where we can get a list of games using this patch?

http://centrorpg.com/index.php?topic=12974 is a game example. Just found out the main community just went down, the MundoRPGMaker. Gotta search for others. However, I got a pretty damn list of system examples for testing out some features (almost every feature).

>Afaik the source is written in assembler (he never finished the C version) which makes it quite useless for us because ASM is hard to understand.

Wow, that's sad :(
However, you're accepting collab, right?

**Comment by fdelapena** [comment]
Thanks, there have been tests on how to detect particular patches and mass game folder scanning between already collected games. If you know RPG Maker 2000/2003 game sites and downloads, there is a [preservation project](https://community.easyrpg.org/t/rpg-maker-2000-preservation-project/199).

> Just found out the main community just went down, the MundoRPGMaker.

Yes, we had at least a link to a game in a bug report pointing to a forum post in that website pointing into a third party file download service. It seems the shutdown was [announced](https://www.facebook.com/mundorpgmaker/photos/a.435909689797509.105289.180581221997025/1288814431173693/?type=3) months ago, sadly.

> However, you're accepting collab, right?

Of course, we have a [contribute section](https://easyrpg.org/contribute/) in the project website for this reason 👍.

**Comment by Ghabry** [comment]
Peperoni found a game that uses DestinyPatch (and aborts because of a script error...): https://rmarchiv.tk/games/889
